mailbox/store/src/main/java/org/apache/james/mailbox/store/probe/MailboxProbe.java

```diff
@@ -54,4 +54,6 @@ ComposedMessageId appendMessage(String username, MailboxPath mailboxPath, InputS
     void reIndexAll() throws Exception;
 
     Collection<String> listSubscriptions(String user) throws Exception;
+
+    void addRights(String namespace, String user, String mailboxName, String targetUser, String rights) throws MailboxException;
```

Comment:
I want to follow on https://github.com/linagora/james-project/pull/985#discussion_r141236276 : 
We mix JMX stuff and interfaces we use pervasively, so we end up with bad API.
IMO, we need a JMX layer that take strings and build objects to call the API (like we do for other protocols)

Reply:
That's another biiiig refactoring, isn't it?

What about splitting to a dedicated ACL probe? Then we can ommit exposing it on JMX, and use types ?